2216/2/bad_words_analyzer.rb

```diff
@@ -0,0 +1,36 @@
+class BadWordsAnalyzer
+  # initialize all criterias for the table
+  def initialize(battles)
+    @battles = battles
+    @round_num = 0
+    @words = divide_into_words
+  end
+
+  def analyze_bad
+    word_size = @words.size
+    bad_words_counter = BadWordsAnalyzer.count_bad_words(@words)
+    @round_num = 1 if @round_num.zero?
+    [@battles.size, bad_words_counter, bad_words_counter / @battles.size,
+     word_size.div(@round_num)]
+  end
+
+  def self.count_bad_words(words)
+    words.select! { |word| word.include?('*') || word.include?('(.') || RussianObscenity.obscene?(word) }.size
+  end
+
+  private
+
+  def divide_into_words
+    words = []
+    @battles.each do |_key, battle_text|
+      @round_num += battle_text.scan(/Раунд [1|2|3][^\s]*/).size
```

Comment:
**Do not mutate state!**
I believe you could rewrite your code without using the `round_num` instance variable.

Reply:
https://ruby-doc.org/core-2.5.1/Enumerable.html